datafusion/physical-expr/src/expressions/binary.rs

```diff
@@ -1689,7 +1812,500 @@ pub fn interval_scalar_interval_op(
             scalar.get_datatype(),
         )))?,
     };
-    Ok(ColumnarValue::Array(ret))
+    Ok(ret)
+}
+
+/// This function handles interval +/- interval operations where the former is
+/// a scalar and the latter is an array, resulting in an interval array.
+pub fn scalar_interval_op_interval(
```

Comment:
cc @tustvold  I am not sure if these exist (yet) in arrow-rs but I think we should be working to port them upstream

Reply:
I intend to spend some time designing what this might look like next week